002-es6-modules.md

```diff
@@ -13,111 +13,165 @@ ES6 modules. At which time this draft would be moved to `ACCEPTED`.
 
-When creating a `DynamicModuleRecord` the list of exports is frozen upon
-construction. No new exports may be added. No exports may be removed.
+When creating a `DynamicModuleRecord` the [`[[Exports]]`](https://tc39.github.io/ecma262/#table-29) is frozen upon construction. No new exports may be added. No exports may be removed. The values of the exports will continue to be mutable however.
 
-## Algorithm
+### 3.1.1 DynamicModuleCreate(O)
 
-When loading a file:
+The abstract operation DynamicModuleCreate with arguments `namespace` is used to allow creation of new DynamicModuleRecords. It performs the following steps:
 
-1. Determine if file is ES or CommonJS (CJS).
-2. If CJS
-  1. Evaluate immediately
-  2. Produce a DynamicModuleRecord from `module.exports`
-3. If ES
-  1. Parse for `import`/`export`s and keep record, in order to create bindings
-  2. Gather all submodules by performing loading dependencies recursively
-    * See circular dep semantics below
-  3. Connect `import` bindings for all relevant submodules (see
-     [ModuleDeclarationInstantiation])
-  4. Evaluate
+1. Let M be a newly created object.
+2. Set M's essential internal methods to the definitions specified in [15.2.1.15 Abstract Module Records](https://tc39.github.io/ecma262/#sec-abstract-module-records)
+3. Set M's [[Namespace]] internal slot to DelegatedModuleNamespaceObjectCreate(`M`, `O`)
+4. Set M's [[Evaluated]] internal slot to **true**
+5. Return M
 
-This still guarantees:
+### 3.2. **DelegatedModuleNamespaceObject**
+
+A ModuleNamespaceObject that performs delegation to an Object when accessing properties. This is used for delegation behavior from CJS `module.exports` when imported by ES modules.
+
+#### Table 1: Internal Slots of DelegatedModuleNamespaceObject Namespace Exotic Objects
+
+Field Name | Value Type | Meaning 
+---| --- | --- 
+[[Delegate]] | Object  | The Object from which to delegate access
+
+#### 3.2.1. `[[Get]] (P, Receiver)`
+
+When the [[Get]] internal method of a module namespace exotic object O is called with property key P and ECMAScript language value Receiver, the following steps are taken:
+
+1. Assert: IsPropertyKey(`P`) is true.
+2. If Type(`P`) is Symbol, then
+3. Return ? OrdinaryGet(`O`,`P`,` Receiver`).
+4. Let exports be the value of `O`'s `[[Exports]]` internal slot.
+5. If `P` is not an element of exports, return **undefined**.
+6. Let m be the value of `O`'s `[[Object]]` internal slot.
+7. If `P` equals **"default"**, return `m`.
+7. Let `value` be ! `O`.`[[Get]]`(`P`,` O`)
+8. Return `value`
+
+#### 3.2.2. `DelegatedModuleNamespaceObjectCreate(module,O)`
+
+The abstract operation DelegatedModuleNamespaceObjectCreate with arguments `O` is used to create a DelegatedModuleNamespaceObject. It performs the following steps:
+
+1. Assert: `module` is a Module Record.
```

Comment:
Records in ECMA262 tend to have a space between the identifier and the word "Record" , see https://tc39.github.io/ecma262/#sec-createimportbinding